sqlite3/provider.go

```diff
@@ -0,0 +1,106 @@
+package sqlite3
+
+import (
+	"database/sql"
+	"fmt"
+
+	"github.com/dhermes/golembic"
+)
+
+const (
+	createTableConstraints = `,
+
+  FOREIGN KEY(previous) REFERENCES revision
+  CHECK (
+    (serial_id = 0 AND previous IS NULL) OR
+    (serial_id != 0 AND previous IS NOT NULL)
+  )
```

Comment:
There is (as of now) nothing that keeps this from drifting when `table.go` changes.